src/hotspot/share/utilities/vmError.cpp

```diff
@@ -818,7 +818,13 @@ void VMError::report(outputStream* st, bool _verbose) {
       st->print(" (0x%x)", _id);                // signal number
       st->print(" at pc=" PTR_FORMAT, p2i(_pc));
       if (_siginfo != nullptr && os::signal_sent_by_kill(_siginfo)) {
-        st->print(" (sent by kill)");
+        if (HandshakeTimedOutThread != p2i(nullptr)) {
+          st->print(" (sent by handshake timeout handler, timed out thread " PTR_FORMAT ")", HandshakeTimedOutThread);
+        } else if (SafepointTimedOutThread != p2i(nullptr)) {
+          st->print(" (sent by safepoint timeout handler, timed out thread " PTR_FORMAT ")", SafepointTimedOutThread);
+        } else {
```

Comment:
I was thinking here that we would only report the timeouts if the current thread was the one we had stashed away. That way we don't hijack a completely different occurrence of SIGILL.